### PR TITLE
OTT-696: Optimisation of account service.

### PR DESCRIPTION
**Pull request by Pubmatic-Dhruv-Sonone**
# Description

This  PR is to optimise account service workflow. The  default account and account config merging part is now moved to fetcher level and the complete merge account config will be  now be cached at cache level.
Added changes to generate event tracker macro template at fetcher level.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: config/macro_processor.go [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+import (
+	"sync"
+
+	"github.com/prebid/prebid-server/macros"
+)
 
 // macroProcessor a global instance which can replace the macros
 // with vaules
 var macroProcessor macros.IProcessor
+var once sync.Once
```
Comment (ShriprasadM) [inline config/macro_processor.go:12]: why we need sync.Once

**Review by ShriprasadM — CHANGES_REQUESTED** [review]

File: config/macro_processor.go (hunk 1 above)
Comment (Pubmatic-Dhruv-Sonone) [reply config/macro_processor.go:12]: To have a singleton instance of macroProcessor. As the template as stored inside the macroProcessor struct, we will only need single instance of the process.

**Review by ShriprasadM — APPROVED** [review]
LGTM

**Comment by Pubmatic-Dhruv-Sonone** [comment]
Closing this pr as changes are already merged to vanilla prebid https://github.com/prebid/prebid-server/pull/2440

**Pull request closed without merging**